include/mbedtls/ssl_internal.h

```diff
@@ -299,6 +299,41 @@ static inline uint32_t mbedtls_ssl_get_input_buflen( const mbedtls_ssl_context *
 #define MBEDTLS_TLS_EXT_SUPPORTED_POINT_FORMATS_PRESENT (1 << 0)
 #define MBEDTLS_TLS_EXT_ECJPAKE_KKPP_OK                 (1 << 1)
 
+/**
+ * \brief        This function checks if the remaining size in a buffer is
+ *               greater or equal than a needed space.
+ *
+ * \param cur    Pointer to the current position in the buffer.
+ * \param end    Pointer to one past the end of the buffer.
+ * \param need   Needed space in bytes.
+ *
+ * \return       Non-zero if the needed space is available in the buffer, 0
```

Comment:
Like I said [in the past](https://github.com/ARMmbed/mbedtls/pull/873#discussion_r110695910), this is contrary to our usual naming / return value convention. Functions that have `check` or similar in the name should return `0` for "everything's fine" and non-zero otherwise. If we want a return value of `0` to mean `false` then we should name the function in a way that clearly indicates the boolean nature of the result, for example `mbedtls_buffer_is_large_enough()`.

Please change either the return value or the name of the function.

Reply:
I can see that it was as you say in Hanno's original PR and I changed it. Sorry about that, didn't know about this convention. I've fixed it in an additional commit.

Reply:
Nothing to be sorry about, that's what reviews are for, and thanks for fixing it :)